src/lib/FreeC/LiftedIR/Syntax/Type.hs

```diff
@@ -44,3 +44,23 @@ data Type
       , typeFreeArg :: Type
       }
  deriving (Eq, Show)
+
+-------------------------------------------------------------------------------
+-- Utility                                                                   --
+-------------------------------------------------------------------------------
+
+-- | Applies a type level expression lifted IR expression to a type constructor.
```

Comment:
Usually you apply a constructor to a type and not the other way around, if that's what is meant here?